src/System.EventLog/src/System/Diagnostics/EventLogInternal.cs

```diff
@@ -0,0 +1,1550 @@
+                    logkey.Close();
+
+                // Revert registry and environment permission asserts
+                CodeAccessPermission.RevertAssert();
+            }
+        }
+
+        private int GetNextEntryPos(int pos)
+        {
+            return pos + IntFrom(cache, pos);
+        }
+
+        private int GetPreviousEntryPos(int pos)
+        {
+            return pos - IntFrom(cache, pos - 4);
+        }
+
+        internal static string GetDllPath(string machineName)
+        {
+            return Path.Combine(SharedUtils.GetLatestBuildDllDirectory(machineName), DllName);
+        }
+
+        private static int IntFrom(byte[] buf, int offset)
+        {
+            // assumes Little Endian byte order.
+            return (unchecked((int)0xFF000000) & (buf[offset + 3] << 24)) | (0xFF0000 & (buf[offset + 2] << 16)) |
+            (0xFF00 & (buf[offset + 1] << 8)) | (0xFF & (buf[offset]));
+        }
+
+        [ComVisible(false)]
+        public void ModifyOverflowPolicy(OverflowAction action, int retentionDays)
+        {
+            string currentMachineName = this.machineName;
+
+            if (action < OverflowAction.DoNotOverwrite || action > OverflowAction.OverwriteOlder)
+                throw new InvalidEnumArgumentException("action", (int)action, typeof(OverflowAction));
+
+            // this is a long because in the if statement we may need to store values as
+            // large as UInt32.MaxValue - 1.  This would overflow an int.
+            long retentionvalue = (long)action;
+            if (action == OverflowAction.OverwriteOlder)
+            {
+                if (retentionDays < 1 || retentionDays > 365)
+                    throw new ArgumentOutOfRangeException(SR.RentionDaysOutOfRange);
+
+                retentionvalue = (long)retentionDays * SecondsPerDay;
+            }
+
+            using (RegistryKey logkey = GetLogRegKey(currentMachineName, true))
+                logkey.SetValue("Retention", retentionvalue, RegistryValueKind.DWord);
+        }
+
+        private void OpenForRead(string currentMachineName)
+        {
+            Debug.WriteLineIf(CompModSwitches.EventLog.TraceVerbose, "EventLog::OpenForRead");
+
+            if (this.boolFlags[Flag_disposed])
+                throw new ObjectDisposedException(GetType().Name);
+
+            string logname = GetLogName(currentMachineName);
+
+            if (logname == null || logname.Length == 0)
+                throw new ArgumentException(SR.MissingLogProperty);
+
+            if (!EventLog.Exists(logname, currentMachineName))        // do not open non-existing Log [alexvec]
+                throw new InvalidOperationException(SR.Format(SR.LogDoesNotExists, logname, currentMachineName));
+            //Check environment before calling api
+            SharedUtils.CheckEnvironment();
+
+            // Clean up cache variables.
+            // [alexvec] The initilizing code is put here to guarantee, that first read of events
```

Comment:
Please remove `[alexvec]` who has probably left the company many years ago!

Reply:
REmoved